### PR TITLE
[feature/crud-context] new permission mechanism implementation.

### PR DESCRIPTION
**Pull request by frknbasaran**

### CODE REVIEW

File: bin/scripts/member_permission_generator.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    if (!members && err) {
+        console.log(err);
+        countlyDb.close();
+        return;
+    }
+
+    function upgrade(member, done) {
+        if (!member.admin_of || !member.user_of) {
+            done();
+        }
```
Comment (mt-sebastien-robert) [inline bin/scripts/member_permission_generator.js:15]: You need a `return` statement here.

Comment (mt-sebastien-robert) [inline bin/scripts/member_permission_generator.js:15]: You need a `return` statement here.

File: bin/scripts/member_permission_generator.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            }
+            done();
+        });
+    };
+
+    async.forEach(members, upgrade, function() {
+        console.log("Finished upgrading member permissions.");
+        countlyDb.close();
+    });
+});
```
Comment (mt-sebastien-robert) [inline bin/scripts/member_permission_generator.js:50]: Linefeed missing ?

**Comment by ar2rsawseen** [comment]
Maybe we can move it to the upgrade script path directly, for next release
bin/upgrade/dev/scripts/member_permission_generator.js

**Pull request merged**